src/main/java/org/graylog2/gelfclient/transport/GelfTcpTransport.java

```diff
@@ -102,13 +102,13 @@ protected void channelRead0(ChannelHandlerContext ctx, ByteBuf msg) throws Excep
 
                             @Override
                             public void channelActive(ChannelHandlerContext ctx) throws Exception {
-                                senderThread.start(ctx.channel());
+                                senderThreadReference.get().start(ctx.channel());
```

Comment:
I think we should use a local `senderThread` variable here instead of the reference. Otherwise might might get a different sender thread back if some other thread updated the reference afterwards.

Reply:
Fixed in https://github.com/Graylog2/gelfclient/pull/43/commits/8ecd202ae4bc32167cb9106e44bc18a91af44a95

Now, the `channelActive` and `channelInactive` callbacks always reference the `senderThread` that was created in that `createBootstrap` invocation.